### PR TITLE
[Tests-Only] Adjust Tags asserts and exceptions for then and given steps

### PR DESCRIPTION
**Pull request by jasson99**
## Description
This PR includes the adjustments in the asserts and exceptions for the then and given steps respectively in the TagsContext file.

## Related Issue
#36810 

## How Has This Been Tested?
CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 1]
```diff
@@ -1342,8 +1343,12 @@ public function theHttpStatusWhenuserRequestsTagsForEntryOwnedByShouldBe(
 		$user, $fileName, $sharingUser, $status
 	) {
 		$this->requestTagsForFile($user, $fileName, $sharingUser);
+		$actualStatus = $this->featureContext->getResponse()->getStatusCode();
 		Assert::assertEquals(
-			$status, $this->featureContext->getResponse()->getStatusCode()
+			$status,
+			$actualStatus,
+			__METHOD__
+			. "Expected status is '$status' but got '$actualStatus'"
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]:
```suggestion
			. " Expected status is '$status' but got '$actualStatus'"
```
There needs to be a space in the string after every use of `__METHOD__` - see PR #36954

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 2]
```diff
@@ -1463,7 +1468,11 @@ public function sharedByHasNoTags($fileName, $sharingUser) {
 		// The array of tags has a single "empty" item at the start.
 		// If there are no tags, then the array should have just this
 		// one entry.
-		Assert::assertCount(1, $tagList);
+		Assert::assertCount(
+			1,
+			$tagList,
+			"Expected no tags, but got '" . \count($tagList) . "' tags"
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]:
```suggestion
			"Expected no tags, but got '" . (\count($tagList) - 1) . "' tags"
```
Or maybe refactor the whole section. Something like:
```
$numTags = \count($tagList) - 1;
Assert::assertEquals(
  0,
  $numTags,
  "Expected no tags for " . $fileName . " but got " . $numTags . " tags"
);
```

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 3]
```diff
@@ -675,12 +675,13 @@ public function theTagHasGroup($type, $tagName, $groups) {
...
-			throw new \Exception(
-				"Expected $count tags, got "
-				. \count($this->requestTagsForUser($user))
-			);
-		}
+		Assert::assertEquals(
+			(int) $count,
+			\count($this->requestTagsforUser($user)),
+			__METHOD__
+			. "Expected $count tags, got "
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]:
```suggestion
			. " Expected $count tags, got "
```

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 1 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/TagsContext.php]: fixed

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 2 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/TagsContext.php]: fixed

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 3 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/TagsContext.php]: fixed

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 4]
```diff
@@ -1463,7 +1468,12 @@ public function sharedByHasNoTags($fileName, $sharingUser) {
 		// The array of tags has a single "empty" item at the start.
 		// If there are no tags, then the array should have just this
 		// one entry.
-		Assert::assertCount(1, $tagList);
+		$numTags = \count($tagList) - 1;
+		Assert::assertCount(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]:
```suggestion
		Assert::assertEquals(
```
IMO this needs to be `assertEquals` now.
https://drone.owncloud.com/owncloud/core/23422/55/13
```
PHPUnit\Framework\Exception: Argument #2 (No Value) of PHPUnit\Framework\Assert::assertCount() must be a countable or iterable
```

**Review by phil-davis — COMMENTED** [review]
You can rebase and squash when fixing this. Then it will be ready to merge.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**